logstash-core/lib/logstash/logging/slow_logger.rb

```diff
@@ -0,0 +1,42 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "logstash/logging/slow_logger/freq_items"
+
+module LogStash
+  module Logging
+
+    class NullLogger
+      def log(threshold, time, data={})
+      end
+      alias_method :warn, :log
+    end
+
+    class SlowLogger
+
+      attr_reader :logger, :freq_items
+
+      def initialize(name="loggers.slow", params={})
+        @logger     = LogStash::Logging::Logger.new(name)
+        @freq_items = LogStash::Logging::Util::FreqItems.new
+      end
+
+      def log(threshold, time, data={})
```

Comment:
`log(threshold, time, data={})`

is `time` millis, nanos? can you add that in the parameter. Maybe change it to `took_millis` or `took_nanos`?